src/org/usfirst/frc/team2399/robot/subsystems/DriveTrain.java

```diff
@@ -4,35 +4,218 @@
+		leftFrontTalon.reverseSensor(RobotMap.REVERSE_LEFT_ENCODER_OUTPUT_CONSTANT);
+		rightFrontTalon.reverseOutput(RobotMap.REVERSE_RIGHT_FRONT_MOTOR_OUTPUT_CONSTANT);
+		rightFrontTalon.reverseSensor(RobotMap.REVERSE_RIGHT_ENCODER_OUTPUT_CONSTANT);
+		
+		/**
+		 * Sets all constants for PID loops
+		 * We won't be using I and D, and they are set to 0 automatically, but
+		 * for clarity and safety's sake, they're set to zero here
+		 * TODO: Set actual values for F and P
+		 */
+		leftFrontTalon.setF(0);
+		leftFrontTalon.setP(0);
+		leftFrontTalon.setI(0);
+		leftFrontTalon.setD(0);
+		
+		rightFrontTalon.setF(0);
+		rightFrontTalon.setP(0);
+		rightFrontTalon.setI(0);
+		rightFrontTalon.setD(0);
+	
+		
 	}
-
-	public void driveLeft(double leftSpeed) {
+	
+	/**
+	 * Velocity - this is better for programmers due to its precision
+	 * Percent - this is better for drivers because it feels more natural
+	 * TODO: Look at p.79 for a closed-loop walkthrough
+	 * TODO: Create a method for resetting calculated control values when
+	 * driver control (percent mode) begins; look through Talon methods first
+	 * TODO: Does Follower mode apply to speed as well as Control Mode?
+	 * @param leftSpeed
+	 */
+	public void driveLeftVelocity(double leftSpeed) {
+		leftFrontTalon.changeControlMode(TalonControlMode.Speed);
+		
+		leftFrontTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
+		leftBackTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
+		leftMiddleTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
+	}
+	
+	public void driveLeftPercent(double leftSpeed) {
+		leftFrontTalon.changeControlMode(TalonControlMode.PercentVbus);
+		
 		leftFrontTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
 		leftBackTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
+		leftMiddleTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
 	}
+	
 
-	public void driveRight(double rightSpeed) {
+	public void driveRightVelocity(double rightSpeed) {
+		rightFrontTalon.changeControlMode(TalonControlMode.Speed);
+		
+		rightFrontTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightBackTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightMiddleTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+	}
+	
+	public void driveRightPercent(double rightSpeed) {
+		rightFrontTalon.changeControlMode(TalonControlMode.PercentVbus);
+		
 		rightFrontTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
 		rightBackTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightMiddleTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
 	}
 
+	
+	
+	/**
+	 * Gets the current position of the robot
+	 * Multiplied by the circumference of the wheel for scaling
+	 * @return
+	 */
+	public double getLeftPosition(){
+		return leftFrontTalon.getPosition() * RobotMap.DRIVETRAIN_WHEEL_CIRCUMFERENCE_CONSTANT;
```

Comment:
Persnickety thing: We know it's a constant because it came from RobotMap and is all caps. the _CONSTANT is redundant.